### DISCUSSION TITLE
'pure' temporary shell environment

### DISCUSSION THREAD

**Discussion by shedali** [pkgxdev/pkgx#436] [Help] [answered]
if i run tea +nodejs.org sh

this is a temporary shell containing the following packages:
curl.se/ca-certs=2023.1.10, unicode.org=71.1.0, openssl.org=1.1.1s, zlib.net=1.2.13, nodejs.org=19.3.0

nice but

echo $PATH:
/Users/franz/.nix-profile/bin:/nix/var/nix/profiles/default/bin:/Users/franz/.tea/unicode.org/v71.1.0/bin:/Users/franz/.tea/unicode.org/v71.1.0/sbin:/Users/franz/.tea/openssl.org/v1.1.1s/bin:/Users/franz/.tea/nodejs.org/v19.3.0/bin:/Users/franz/.rd/bin:/Users/franz/.bun/bin:/Users/franz/Library/pnpm:/Users/franz/.tea/tea.xyz/v*/bin:/Users/franz/Library/Android/sdk/tools/bin/:/Users/franz/.fastlane/bin:/usr/local/bin:/usr/local/sbin:/opt/homebrew/opt/openssl@1.1/bin:/usr/local/opt/curl/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/System/Cryptexes/App/usr/bin:/usr/bin:/bin:/usr/sbin:/sbin:/usr/local/MacGPG2/bin:/Library/Apple/usr/bin:/Users/franz/.nix-profile/bin:/nix/var/nix/profiles/default/bin:/Users/franz/go/bin:/Users/franz.fnm/current/bin:/Users/franz/.cargo/bin:/Users/franz/.deno/bin:/Users/franz/.pyenv:/Users/franz/bin:/Users/franz/lib/vsts-cli/bin:/usr/local/opt/asdf/installs/python/3.8.5/bin:/usr/local/go/bin:/usr/local/opt/python/libexec/bin:/usr/local/opt/asdf/installs/python/3.7.1/bin:/Users/franz/.local/bin:/Users/franz/.fzf/bin:/Users/franz/dev/**/platform-fe/flutter/bin:/Users/franz/n/bin:/Users/franz/.npm/bin:/Users/franz/pnpm-home:/Users/franz/Library/Python/3.8/bin:/Users/franz/Library/Python/2.7/bin/usr/local/opt/python@3.9/bin:/Users/franz/.pulumi/bin

and shell has access to binaries available on PATH including homebrew installs.

How can I avoid this?

This may be related to the tea --strain idea at https://github.com/teaxyz/cli/discussions/362

(original message discord https://discord.com/channels/906608167901876256/1082864420352426047/1082864420352426047)

**Comment by jhheider** [comment, marked as answer]
If you need it, this appears to do what you're asking:
```sh
$ PATH=$(which tea) tea +nodejs.org sh
this is a temporary shell containing the following packages:
curl.se/ca-certs=2023.1.10, unicode.org=71.1.0, openssl.org=1.1.1s, zlib.net=1.2.13, nodejs.org=19.7.0
when done type: `exit'
tea ~ echo $PATH
/Users/jacob/.tea/unicode.org/v71.1.0/bin:/Users/jacob/.tea/unicode.org/v71.1.0/sbin:/Users/jacob/.tea/openssl.org/v1.1.1s/bin:/Users/jacob/.tea/nodejs.org/v19.7.0/bin:/Users/jacob/.tea/tea.xyz/v0.24.9/bin:/usr/local/bin/tea
```

**Reply by shedali** [reply]
awesome thanks!

**Comment by mxcl** [comment]
We would consider a flag to remove inherited `env` (suggestions welcome).

Notably we haven't packaged all of POSIX yet so this will likely leave a somewhat broken shell.
